src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -108,38 +108,45 @@ public void Start()
             stopping = false;
             pipelineInvocationTasks = new ConcurrentDictionary<Task, Task>();
 
-            
-            for(var i = 0; i < numberOfClients; i++)
+            var exceptions = new ConcurrentQueue<Exception>();
+            Parallel.For(0, numberOfClients, i =>
```

Comment:
@ramonsmits you can await a task more then once, it just returns the result the second time, no need for lazy